crates/evm/src/execute.rs

```diff
@@ -287,7 +280,38 @@ where
     pub(crate) ctx: F::ExecutionCtx<'a>,
     pub(crate) parent: &'a SealedHeader<HeaderTy<N>>,
     pub(crate) assembler: Builder,
-    pub(crate) simulated: bool,
+}
+
+/// Conversions for executable transactions.
+pub trait ExecutorTx<B: BlockExecutor> {
+    /// The inner signed transaction.
+    type T;
+
+    /// Converts the transaction into [`ExecutableTx`].
+    fn as_executable<'a>(&'a self) -> impl ExecutableTx<B> + 'a
```

Comment:
so I feel like just returning `impl ExecutableTx<Builder::Executor>` should work here